docs/src/osdk/reference/manifest.md

```diff
@@ -10,48 +10,76 @@ If there is only one crate and no workspace,
-the `OSDK.toml` of the workspace will be used
-- If running commands in the crate directory
-  - If the crate has `OSDK.toml`,
-  the `OSDK.toml` of the crate will be used.
-  - Otherwise, the `OSDK.toml` of the workspace will be used.
+OSDK will firstly refer to the crate-level manifest, then
+query the workspace-level manifest for undefined fields.
+In other words, missing fields of the crate manifest
+will inherit values from the workspace manifest.
 
 ## Configurations
 
 Below, you will find a comprehensive version of
-the available configuration options in the manifest.
+the available configuration tree in the manifest.
 
-```toml
-kcmd_args = ["init=/bin/busybox", "path=/usr/local/bin"] # <1>
-init_args = ["sh", "-l"] # <2>
-initramfs="./build/initramfs.cpio.gz" # <3>
+Here are notes for some fields with special value treatings:
+ - `*` marks the field as "will be evaluated", that the final
+value of string `"S"` will be the output of `echo "S"` using the
+host's shell.
+ - `+` marks the path fields. The relative paths written in the
+path fields will be relative to the manifest's enclosing directory.
 
-[boot]
-loader = "grub" # <4>
-protocol = "multiboot2" # <5>
-grub-mkrescue = "/usr/bin/grub-mkrescue" # <6>
-ovmf = "/usr/bin/ovmf" # <7>
+If values are given in the tree that's the default value inferred
+if that the field is not explicitly stated.
 
-[qemu]
-path = "/usr/bin/qemu-system-x86_64" # <8>
-machine = "q35" # <9>
-args = [ # <10>
-    "-enable-kvm",
-    "-m 2G", 
-    "-device virtio-keyboard-pci,disable-legacy=on,disable-modern=off"
-] 
+```
+project_type             # The type of the current crate. Can be lib/kernel[/module]
+
+# --------------------------- the default schema settings ---------------------
+vars = []                # List of lists. These are the env vars that will be set before
+                         # any evaluation happens. The variables will be evaluated
+                         # **SEQUENTIALLY**.
+                         # The reserved variables:
+                         #  - `OSDK_CWD`: The directory of the OSDK manifest.
+build
+|- features = []         # List of strings, the same as Cargo
+|- profile = "dev"       # String, the same as Cargo
+boot
+|- method                # "grub-rescue-iso"/"qemu-direct"/"grub-qcow2"
+|- kcmd_args             # <1>
+|- init_args             # <2>
+|- initramfs +           # The path to the initramfs
+grub                     # Grub options are only needed if boot method is related to GRUB
+|- mkrescue-path +       # The path to the `grub-mkrescue` executable
+|- protocol              # The protocol GRUB used. "linux"/"multiboot"/"multiboot2"
+qemu
+|- path +                # The path to the QEMU executable
+|- args *                # String. <3>
+run                      # Special settings for running, which will override default ones
+|- vars                  # The run specific variables evaluate after .vars
+|- build                 # Overriding .build
```

Comment:
I want to keep the design since run is an action that can be configured to override settings for the build action. What you suggest breaks the hierarchy and introduce inconsistencies since put features and profile settings to the root level is not appropriate.